File: 0069-sqrtx/0069-sqrtx.py
class Solution:
    def mySqrt(self, x: int) -> int:
        lo = 0
        hi = x
        res = 0
        while lo <= hi:
            m = lo + ((hi - lo) // 2)
            if m ** 2 > x:
                hi = m - 1
            elif m ** 2 < x:
                lo = m + 1
                res = m
            else:
                return m
        return res
            

File: 0069-sqrtx/0069-sqrtx.cpp
class Solution {
public:
    int mySqrt(int x) {
        int l = 0;
        int r = x;
        int res = 0;
        while(l <= r){
            long m = l + ((r - l) / 2);
            long sq = m * m;
            if(sq > x){
                r = m - 1;
            }
            else if(sq < x){
                l = m + 1;
                res = m;
            }
            else{
                return m;
            }
        }
        return res;
    }
};